src/Raven.Server/Documents/AI/ChatCompletionClient.cs

```diff
@@ -145,9 +146,10 @@ public async Task<AiResponse> CompleteAsync(JsonOperationContext context, HttpRe
         if (string.IsNullOrEmpty(content))
         {
             choice0.TryGet(Constants.ResponseFields.FinishReason, out string finishReason);
-            choice0.TryGet(Constants.ResponseFields.Refusal, out string refusal); 
+            _ = choice0.TryGet(Constants.ResponseFields.Refusal, out string refusal) || msg.TryGet(Constants.ResponseFields.Refusal, out refusal);
+
             //TODO: full output if we get here?
-            throw new RefusedToAnswerException("The request was refused by the model")
+            throw new RefusedToAnswerException($"The request was refused by the model: '{refusal}'")
```

Comment:
this is in general fine'ish, but imagine if you would have that exception in 10-20 places? In that case it is better to create a factory pattern and make sure that it behaves exactly the same everywhere, not only in this one place. No need to change it here, this is a tip for future. We are doing that e.g. in IndexDoesNotExistException